docs/how-to/push-update-on-mainnet.md

```diff
@@ -0,0 +1,118 @@
+Overview
+========
+
+Software update is a protocol mechanism that allows nodes to
+automatically update their executables without changing protocol
```

Comment:
A minor nit: the binary substitution mechanism is powerful enough that any changes at all can be introduced at all, so the "without changing the protocol itself" is merely about intent, I'd say.  Perhaps that should be clarified.

Reply:
I'll update it.